### PR TITLE
observe-hook: add new thread to leave graphs upon removing a group

### LINKED ISSUE #3857: landscape: leaving a group dumps all channels, even if unjoined, into home workspace
**Describe the bug**
Click on the _Settings_ and _Refresh_ icons produce errors and incorrect function calls when accessing channels I have never joined from groups I am no longer a member of.

**To Reproduce**
1. Join a group by clicking on a link in a chat
2. Click on each channel in the group but do **not** join any of the channels in that group
3. Go to landscape home
4. Click on the newly created group tile 
5. Leave the group via button
6. Go to landscape home
4. Click on your Home tile
5. You should see the channels from that group greyed out in your channel list
6. Click on one of these channels on the LHS menu

**Expected behavior**
These channels should not appear in the list under 'home' as I never joined them and I left the group.

**Actual behavior**
With a channel selected after steps above
1. Clicking on the 'cog settings icon' will result in an error screen "“Cannot read property 'metadata' of undefined”"
2. Clicking on the 'refresh icon' will result the in 'new DM' modal appearing

**Screenshots**
Screen Shot 2020-11-02 at 12 28 29 AM
(clicking on the refresh icon)

**Desktop (please complete the following information):**
 - OS: MacOS v10.15.7
 - Browser Chrome

### PR DESCRIPTION
**Pull request by tacryt-socryp**
Fixes #3857 and #2992. Still needs testing, this is just a sketch. Will finish it up later today.

Edit: tested and ready to go

### CODE REVIEW

**Comment by tacryt-socryp** [comment]
This has been tested to work properly. Thanks!

**Comment by tacryt-socryp** [comment]
Note: back to WIP status now that I'm working on resolving both metadata cleanup and graph archiving.

**Comment by tacryt-socryp** [comment]
Note: This is ready to go. Tested.

**Comment by tacryt-socryp** [comment]
I've made it even better now: we now clean up all contacts data, group-push-hook, and group-pull-hook as well. Tested and it works great.

**Review by liam-fitzgerald — APPROVED** [review]

**Comment by vvisigoth** [comment]
Will all of this group leaving stuff include actually cleaning up old DMs
correctly
—
~poldec-tonteg
http://urbit.org

**Comment by matildepark** [comment]
Old DMs don't have an archived flag; our only option would really be to delete them. What I've advised @liam-fitzgerald is to incorporate the cleanup in Chat migration so that old DMs with no participants are treated as archived graphs, and our DM creation flow skips over archived graphs.

**Comment by vvisigoth** [comment]
> Old DMs don't have an archived flag; our only option would really be to
delete them. What I've advised @liam-fitzgerald
<https://github.com/liam-fitzgerald> is to incorporate the cleanup in Chat
migration so that old DMs with no participants are treated as archived
graphs, and our DM creation flow skips over archived graphs.

By that logic, couldn't he DM creation flow just skip over DMs with no
participants?
—
~poldec-tonteg
http://urbit.org

**Comment by liam-fitzgerald** [comment]
With current chat, we can't skip over empty DMs because there might be a naming conflict. Our best with chat-store is selfdestructing DMs when a counterparty leaves. Likewise with graph store, but archiving them instead.

**Comment by vvisigoth** [comment]
> there might be a naming conflict

Why not add a ran to all future DMs. Does it matter?
—
~poldec-tonteg
http://urbit.org

**Comment by matildepark** [comment]
We check to see who’s hosting the chat by matching the ship names programmatically so by adding random numbers we just add more edge cases and don’t solve the problem. It would pick the first number if the regex was loose.

**Comment by matildepark** [comment]
Consider taking this to #3826 or #3926 (related issues, but different aspects of the problem).

File: pkg/arvo/app/contact-view.hoon [hunk 1]
```diff
@@ -181,22 +181,7 @@
...
-          (group-pull-poke %remove rid)
-      ==
-    ;:  weld
-      group-pokes
-      :~  (contact-hook-poke [%remove path.act])
-          (group-poke [%remove-group rid ~])
-          (contact-poke [%delete path.act])
-      ==
-      (delete-metadata path.act)
-    ==
```
Comment (matildepark) [inline pkg/arvo/app/contact-view.hoon:199]: @ixv @liam-fitzgerald Does this impact #3604? I believe we wanted a freeze on all code touching that PR as per @philipcmonk. Should we revert this PR back out of next-userspace if so?

Comment (tacryt-socryp) [reply pkg/arvo/app/contact-view.hoon:199]: We could remove the modification to %contact-view without harming anything.

Comment (liam-fitzgerald) [reply pkg/arvo/app/contact-view.hoon:199]: It should be fine, but I'll let @ixv confirm

Comment (ixv) [reply pkg/arvo/app/contact-view.hoon:199]: I don't believe so, but I'm about to do another round of testing so I will try it out.

Comment (ixv) [reply pkg/arvo/app/contact-view.hoon:199]: just finished testing, its all good

**Pull request merged**